tests/test_single_topology.py

```diff
@@ -1217,3 +1246,286 @@ def test_hif2a_plot_force_constants():
         all_axes[3].set_ylim(0, 1)
 
         plt.show()
+
+
+@pytest.mark.nocuda
+def test_hif2a_pairs_setup_st():
+    """
+    Test that we can setup all-pairs single topology objects in hif2a.
+    """
+    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
+        mols = read_sdf(path_to_ligand)
+
+    pairs = [(mol_a, mol_b) for mol_a in mols for mol_b in mols]
```

Comment:
nit: This does N^2, including self-self transformations, not sue this is informative. Might be better to do upper triangular of the pairs

Reply:
intentional - I wanted to test setting up the single topology both ways `SingleTopology(mol_a, mol_b,...)` and `SingleTopology(mol_b, mol_a, ...)`